src/helpers.js

```diff
@@ -33,3 +36,17 @@ export function arraysEqual(a, b) {
 
   return a.every((element, index) => element === b[index]);
 }
+
+function memoizeString<T>(fn: (string) => T): (string) => T {
```

Comment:
StringManipulator should also be generic. I think I'll do something like this:

```js
// 1 stands for 1 argument
type function1<I, O> = (I) => O
type higherOrderFunction1<I, O> = (function1<I, O>) => function1<I, O>
const memoizeString<T>: higherOrderFunction1<string, T> = ...
```

maybe with `HOF1` instead of `higherOrderFunction1`